flask_lastuser/__init__.py

```diff
@@ -218,15 +232,24 @@ def before_request(self):
                 user = self.load_user(g.lastuser_cookie['userid'])
                 if not user:
                     # As above, try to create user record
-                    user = self.lastuser.login_from_cookie(g.lastuser_cookie['userid'])
+                    user = self.lastuser.login_from_cookie(
+                        g.lastuser_cookie['userid'])
+        return user, user_from_token, token_error
+
+    def before_request(self):
+        """
+        Listener that is called at the start of each request. Responsible for
+        """
+        user, user_from_token, token_error = self.get_logged_in_user()
 
         if not user:
             g.lastuser_cookie.pop('userid', None)
             g.lastuser_cookie.pop('sessionid', None)
 
         g.user = user
+        lastuser_cache = self.get_lastuser_cache(header_only=True)
```

Comment:
You've already called this method in the `token_auth` method, so it's now going to hit cache twice in the same request.

Reply:
Return it from the chain of method calls instead. Don't hit cache for in-thread data, and don't make cache a mandatory requirement.